src/main/java/com/sap/oss/phosphor/fosstars/data/AbstractStaticScanToolsDataProvider.java

```diff
@@ -199,38 +200,32 @@ private static boolean scanGitHubAction(Map<?, ?> githubAction,
         .filter(Iterable.class::isInstance)
         .map(Iterable.class::cast)
         .map(jobs -> scanJobs(jobs, matchers))
-        .orElse(false);
+        .orElse(Optional.empty());
   }
 
   /**
    * Search for the given predicate matches in the given list of action jobs.
    *
    * @param jobs Iterable list of GitHub action jobs.
    * @param matchers predicates to match through the given action.
-   * @return True if one of the predicate finds matches. False otherwise.
+   * @return Optional<{@link Map}> step if one of the predicate finds matches.
+   *         Optional.empty otherwise.
    */
-  private static boolean scanJobs(Iterable<?> jobs, Map<String, Predicate<String>> matchers) {
+  private static Optional<Map> scanJobs(Iterable<?> jobs, Map<String, Predicate<String>> matchers) {
     return IteratorUtils.toList(jobs.iterator()).stream()
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .filter(job -> job.containsKey("steps"))
         .filter(Map.class::isInstance)
         .map(Map.class::cast)
         .map(job -> job.get("steps"))
-        .filter(Iterable.class::isInstance)
-        .map(Iterable.class::cast)
-        .anyMatch(steps -> scanSteps(steps, matchers));
-  }
-
-  /**
-   * Search for the given predicate matches in the given list of job steps.
-   *
-   * @param jobs Iterable list of GitHub job steps.
-   * @param matchers predicates to match through the given action.
-   * @return True if one of the predicate finds matches. False otherwise.
-   */
-  private static boolean scanSteps(Iterable<?> steps, Map<String, Predicate<String>> matchers) {
-    return IteratorUtils.toList(steps.iterator()).stream()
+        .filter(ArrayList.class::isInstance)
+        .map(ArrayList.class::cast)
+        .flatMap(step -> step.stream())
         .filter(Map.class::isInstance)
         .map(Map.class::cast)
-        .anyMatch(step -> MATCH_MAP_PREDICATE.test(step, matchers));
+        .filter(step -> MATCH_MAP_PREDICATE.test((Map<?, ?>) step, matchers))
+        .findAny();
```

Comment:
@ManjunathMS35 I think, I have fixed it. Please review it and check if we can optimize this.

Reply:
@sourabhsparkala The solution looks good to me. If there is nothing to change in this PR could please approve it?